### PR TITLE
129 - Dataset Action Buttons UI

### LINKED ISSUE #129: [Spike - Frontend] UI of the Dataset page action buttons
## Overview of the Feature Request

Spike issue of the UI work for #128.

The goal is to develop just the UI logic, without any behavior that requires data access.

### Mockup of the requested element

dataset_actionbuttons_1

dataset_actionbuttons_2

## What inspired the request?

- #128 

## What existing behavior do you want changed?

N/A

## Any brand new behavior do you want to add to Dataverse?

N/A

## Any open or closed issues related to this feature request?

- #128

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR adds the Dataset Action Buttons UI to the Dataset Page View Mode

## Which issue(s) this PR closes:

- Closes #129 

## Special notes for your reviewer:
Some dataset properties are not connected to js-dataverse yet, so take that into account that if you test the e2e. For example the LinkDatasetButton is always going to appear in the e2e page, but if you look at the unit tests the LinkDatasetButton will only appear if the Dataset is released.

### Changes in this PR
1. Dataset Action Buttons added to the UI
2. New properties added to the Dataset model
3. DatasetProvider added to get the dataset from any component in the Dataset Page
4. Files components refactored to read the new Dataset properties

## Suggestions on how to test this:

To test the changes introduced by this PR, follow these steps:

1. Install the necessary dependencies using `npm install`.
5. Build the design system with `cd packages/design-system && npm run build`.
6. Navigate back to the root directory with `cd ../../`.
7. Start Storybook with `npm run storybook`.
8. Visit the [Dataset Action Buttons section in the Storybook](http://localhost:6006/?path=/story/sections-dataset-page-datasetactionbuttons--with-publish-permissions)

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
image

## Is there a release notes update needed for this change?:
Dataset Action Buttons added to the Dataset Page

### CODE REVIEW

**Comment by GPortas** [comment]
It seems that the top of the citation block is not aligned with the top of the buttons. Can we fix this?

allignmentbuttons

**Review by GPortas — COMMENTED** [review]
Looks good!

It seems that the top of the citation block is not aligned with the top of the buttons. Can we fix this?

allignmentbuttons

File: src/dataset/domain/models/Dataset.ts [hunk 1]
```diff
@@ -234,13 +262,40 @@ export class Dataset {
     public readonly labels: DatasetLabel[],
     public readonly summaryFields: DatasetMetadataBlock[],
     public readonly license: DatasetLicense,
-    public readonly metadataBlocks: DatasetMetadataBlocks
+    public readonly metadataBlocks: DatasetMetadataBlocks,
+    public readonly permissions: DatasetPermissions,
```
[the remark this answers is not in the document]
Comment (MellyGray) [reply src/dataset/domain/models/Dataset.ts:266]: Exactly like that, in order to create the Dataset you'll need the user permissions.

I understand your concern and these are the reasons to have them together:

1. The dataset permissions are always used with the dataset model. So I think these permissions belong to the Dataset model, for keeping the cohesion
2. If we have another page where we want to use the dataset permissions without the model. We can do it, getDatasetPermissions and you return the DatasetPermissions interface. But this is not going to happen in the Dataset Page, in the dataset page we always want the dataset and the dataset permissions, so it makes sense to have them together.
3. I separated the File Permissions from the File model and it's being a nightmare, you need to remember to call these permissions when you get the files, initialise the React context, wait for them to be fetched, the tests are much more complex. Too much complexity for the aim of keeping the separation of concerns, when in reality I never use the file permissions without the files itself. I want to refactor that but I was also waiting for some feedback on this other approach.

I think the Dataset model is in reality the DatasetPage model, so its aim is to represent a DatasetPage, and the dataset page doesn't make sense without the permissions.

Very interesting conversation, please let me know if you can think of some problems of this approach

**Comment by MellyGray** [comment]
@GPortas I applied the requested changes

1. Show action buttons on the Default page of the storybook. Only the Access Dataset one because it's more realistic
2. Align the top of the buttons with the citation block
3. Fix the publication disabled on the Draft with all Dataset Permissions story

**Review by GPortas — APPROVED** [review]
LGTM!

**Pull request merged**